lib/irc/formatting.js

```diff
@@ -45,15 +45,21 @@ var colorCodesToHtmlNames = {};
 var htmlNames = Object.keys(htmlNamesToColorCodes);
 htmlNames.forEach(function(htmlName) {
     htmlNamesToColorCodes[htmlName].forEach(function(colorCode) {
-        colorCodesToHtmlNames[colorCode] = htmlNamesToHex[htmlName];
+        colorCodesToHtmlNames[colorCode.substr(1)] = htmlNamesToHex[htmlName];
     });
 });
 
-var STYLE_BOLD = '\u0002';
-var STYLE_ITALICS = '\u001d';
-var STYLE_UNDERLINE = '\u001f';
-var STYLE_CODES = [STYLE_BOLD, STYLE_ITALICS, STYLE_UNDERLINE];
-var RESET_CODE = '\u000f';
+const STYLE_BOLD = '\u0002';
+const STYLE_ITALICS = '\u001d';
+const STYLE_UNDERLINE = '\u001f';
+const STYLE_CODES = [STYLE_BOLD, STYLE_ITALICS, STYLE_UNDERLINE];
+const RESET_CODE = '\u000f';
+const REVERSE_CODE = '\u0016';
+const STYLE_DEFAULT_STATE = {
+    "color": null,
+    "bcolor": null,
+    "history": []
```

Comment:
You need to go into much more detail about:
 - What will `history` contain? When will things get added/removed from it?
 - What will `color` and `bcolor` be? What does `null` represent?